processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -565,6 +566,32 @@ public Integer getDimensionIndex(String dimension)
     return dimensionOrder.get(dimension);
   }
 
+  public LinkedHashMap<String, Integer> getDimensionOrder()
+  {
+    return dimensionOrder;
+  }
+
+  public void loadDimensionOrder(LinkedHashMap<String, Integer> oldDimensionOrder)
```

Comment:
can this method be called at any point or only before ingesting any row ? 
If this needs to be called before ingesting any row, can we add a sanity check to ensure its called before add is called ? maybe check size of index = 0 here.